File: src/main/kotlin/mateuszmacholl/formica/validation/post/channelOrCoordinatesNotNull/ChannelOrCoordinatesPassed.java
package mateuszmacholl.formica.validation.post.channelOrCoordinatesNotNull;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Constraint(validatedBy = OneOfThemNotNullValidator.class)
@Target({ ElementType.TYPE })
@Retention(RetentionPolicy.RUNTIME)
public @interface ChannelOrCoordinatesPassed {
    String message() default "Pass channel or coordinates";
	Class<?>[] groups() default {};
	Class<? extends Payload>[] payload() default {};
    String channel();
	String coordinates();
}

File: src/main/kotlin/mateuszmacholl/formica/validation/post/channelOrCoordinatesNotNull/ChannelOrCoordinatesPassed.kt
package mateuszmacholl.formica.validation.post.channelOrCoordinatesNotNull

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@Constraint(validatedBy = [ChannelOrCoordinatesPassedValidator::class])
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
annotation class ChannelOrCoordinatesPassed(
        val message: String = "Pass channel or coordinates",
        val groups: Array<KClass<*>> = [],
        val payload: Array<KClass<out Payload>> = [],
        val channel: String, val coordinates: String)